File: src/algorithms/is-fascinating.js
/**
 * Given an integer n that consists of exactly 3 digits, returns true if n is
 * fascinating, or false otherwise.
 * 
 * A number is fascinating if, after concatenating n with the numbers 2 * n and 
 * 3 * n, the resulting number contains all the digits from 1 to 9 exactly once 
 * and does not contain any 0's.
 * 
 * @param {number} n - A 3-digit number
 * @returns {boolean} - Returns true if n is a fascinating number
 */
function isFascinating (n) {
  let numSet = new Set();
  let num = String(n);
  num = num.concat(n * 2, n * 3);
  for (const digit of num) {
    if (digit === '0' || numSet.has(digit)) {
      return false;
    } else {
      numSet.add(digit);
    }
  }
  return numSet.size === 9;
}

File: src/algorithms/is-fascinating.py
def isFascinating(n):
    """You are given an integer n that consists of exactly 3 digits. We call 
    the number n fascinating if, after the following modification, the 
    resulting number contains all the digits from 1 to 9 exactly once and does 
    not contain any 0's:
    • Concatenate n with the numbers 2 * n and 3 * n.

    Return true if n is fascinating, or false otherwise.

    Args:
        n (int): A positive integer.

    Returns:
        Boolean: Returns True if the resulting concatenation is fascinating.
    """
    res = f"{n}"
    res += str(n * 2) + str(n * 3)
    number_list = [int(digit) for digit in str(res)]
    number_set = set()
    for num in number_list:
        if (num == 0 or num in number_set):
            return False
        number_set.add(num)
    return len(number_set) == 9
